src/main/java/net/minestom/server/utils/chunk/ChunkUtils.java

```diff
@@ -81,4 +83,43 @@ public static Chunk retrieve(Instance instance, Chunk originChunk, double x, dou
     public static Chunk retrieve(Instance instance, Chunk originChunk, Point position) {
         return retrieve(instance, originChunk, position.x(), position.z());
     }
+
+    /**
+     * Computes the effective view distance based on the client's view distance and the target instance's view distance.
+     * The effective view distance is the minimum of the client's view distance and the instance's view distance, plus one.
+     *
+     * <p>Examples:
+     * <ul>
+     *   <li>Client VD = 10, Instance VD = 16 → min(10, 16) + 1 = 11</li>
+     *   <li>Client VD = 16, Instance VD = 8 → min(16, 8) + 1 = 9</li>
+     *   <li>Client VD = 12, Instance = null → min(12, ServerFlag.CHUNK_VIEW_DISTANCE) + 1</li>
+     * </ul>
+     *
+     * @param clientViewDistance the client's view distance setting
+     * @param targetInstance the target instance, or null to use the default server view distance
+     * @return the effective view distance
+     */
+    public static int computeEffectiveViewDistance(byte clientViewDistance, @Nullable Instance targetInstance) {
```

Comment:
i don't think it makes much sense to use a byte as an input. i understand it's to match the byte stored in `ClientSettings`, but this method isn't related to that, this is just a helper that could be used by users. additionally, it returns an int anyway, so i don't think passing a byte just to match the `ClientSettings` is appropriate here. ~`ClientSettings.viewDistance()` can be cast to an int without any problems.~ byte is implictly widened by the compiler, so this won't even need any other changes (apart from removing the explicit byte casts elsewhere).
```suggestion
    public static int computeEffectiveViewDistance(int clientViewDistance, @Nullable Instance targetInstance) {
```